modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs

```diff
@@ -176,66 +176,183 @@ private void _ExportBeginImpl(string[] features, bool isDebug, string path, long
+                    recurseDir: dir =>
+                    {
+                        if (ridOS == OS.DotNetOS.iOS)
+                        {
+                            // don't recurse into dsym folders
+                            return !dir.EndsWith(".dsym", StringComparison.InvariantCultureIgnoreCase);
+                        }
+                        return true;
+                    },
+                    addEntry: (p_path, p_isFile) =>
+                    {
+                        // we get called back for both directories and files, but we only package files for now
+                        if (p_isFile)
+                        {
+                            if (embedBuildResults)
+                            {
+                                var filePath = SanitizeSlashes(Path.GetRelativePath(publishOutputTempDir, p_path));
+                                var fileData = File.ReadAllBytes(p_path);
+                                var hash = Convert.ToBase64String(SHA512.HashData(fileData));
+
+                                manifest.Append($"{filePath}\t{hash}\n");
+
+                                AddFile($"res://.godot/mono/publish/{arch}/{filePath}", fileData, false);
+                            }
+                            else
+                            {
+                                if (ridOS == OS.DotNetOS.iOS && p_path.EndsWith(".dat"))
+                                {
+                                    AddIosBundleFile(p_path);
+                                }
+                                else
+                                {
+                                    AddSharedObject(p_path, tags: null,
+                                        Path.Join(projectDataDirName,
+                                            Path.GetRelativePath(publishOutputTempDir, Path.GetDirectoryName(p_path))));
+                                }
+                            }
+                        }
+                    });
+            }
+        }
+
+        private static void RecursePublishContents(string path, Func<string, bool> filterDir,
+            Func<string, bool> filterFile, Func<string, bool> recurseDir,
+            Action<string, bool> addEntry)
+        {
+            foreach (string file in Directory.GetFiles(path, "*", SearchOption.TopDirectoryOnly))
+            {
+                if (filterFile(file))
+                {
+                    addEntry(file, true);
+                }
+            }
+
+            foreach (string dir in Directory.GetDirectories(path, "*", SearchOption.TopDirectoryOnly))
+            {
+                if (filterDir(dir))
+                {
+                    addEntry(dir, false);
+                }
+                else if (recurseDir(dir))
                 {
-                    var fileData = Encoding.Default.GetBytes(manifest.ToString());
-                    AddFile($"res://.godot/mono/publish/{arch}/.dotnet-publish-manifest", fileData, false);
```

Comment:
I think the `.dotnet-publish-manifest` was used by the Android exports, but I was able to export to Android and the apk worked fine, so not sure. We do still look it up in `godotsharp_dirs.cpp`.

Reply:
This got moved further down, to a recursive directory listing with callbacks, instead of a straight recursive list of files. The existing behaviour should be the same, in the end.

We need to filter out whole directory trees when we look for files to add to the package, because the nativeaot artifacts produce a bunch of directories with the same filenames in them, so it's very hard to filter those out on a filename basis. The straight recursive file listing would make for complicated logic. It's easier to list directories first, determine which ones we want, and then list files for packaging on those.

Reply:
I only see the `manifest.Append`, but not the `manifest.ToString`.

Reply:
This got lost in the shuffle, turns out, it's restored now.